components/checkout/CheckoutSidebarView/CheckoutSidebarView.tsx

```diff
@@ -10,18 +10,29 @@ import useCheckout from '@framework/checkout/use-checkout'
 import ShippingWidget from '../ShippingWidget'
 import PaymentWidget from '../PaymentWidget'
 import s from './CheckoutSidebarView.module.css'
+import { useCheckoutContext } from '../context'
 
 const CheckoutSidebarView: FC = () => {
+  const [loadingSubmit, setLoadingSubmit] = useState(false)
   const { setSidebarView, closeSidebar } = useUI()
-  const { data: cartData } = useCart()
+  const { data: cartData, revalidate: refreshCart } = useCart()
   const { data: checkoutData, submit: onCheckout } = useCheckout()
+  const { clearCheckoutFields } = useCheckoutContext()
 
   async function handleSubmit(event: React.ChangeEvent<HTMLFormElement>) {
-    event.preventDefault()
+    try {
+      setLoadingSubmit(true)
+      event.preventDefault()
 
-    await onCheckout()
-
-    closeSidebar()
+      await onCheckout()
+      clearCheckoutFields()
```

Comment:
@goncy I added this which clears the checkout context after a successful checkout.

I was wondering if the `clearCheckoutFields` function should be returned from the `useCheckout` hook directly (like the `submit` function) so that we never have to interact with **both** that and `useCheckoutContext`.

But I decided not to go for this because I noticed that it would probably have to be returned as a "mutation" according to the type (like `submit` currently is) which isn't exactly correct semantically. 

```
export type GetCheckoutHook<T extends CheckoutTypes = CheckoutTypes> = {
  data: T['checkout'] | null
  input: {}
  fetcherInput: { cartId?: string }
  swrState: { isEmpty: boolean }
  mutations: { submit: UseSubmitCheckout }   <--- could add `clearCheckoutFields` here
}
```

Let me know what you think.

Reply:
Clearing the checkout fields there seems ok, mutations usually mutates or interacts with the data it manages, in this case is just a side effect, I preffer to leave it in its own place but we can get back at it later if it starts to feel weird

Reply:
@goncy anything else to address in this PR?

Reply:
Hey @cpv123 can you please resolve the current conflicts? Thanks!